src/NzbDrone.Core/Indexers/Newznab/NewznabRequestGenerator.cs

```diff
@@ -9,21 +9,79 @@ namespace NzbDrone.Core.Indexers.Newznab
 {
     public class NewznabRequestGenerator : IIndexerRequestGenerator
     {
-        public Int32 MaxPages { get; set; }
-        public Int32 PageSize { get; set; }
+        private readonly INewznabCapabilitiesProvider _capabilitiesProvider;
+        public int MaxPages { get; set; }
+        public int PageSize { get; set; }
         public NewznabSettings Settings { get; set; }
 
-        public NewznabRequestGenerator()
+        public NewznabRequestGenerator(INewznabCapabilitiesProvider capabilitiesProvider)
         {
+            _capabilitiesProvider = capabilitiesProvider;
+
             MaxPages = 30;
             PageSize = 100;
         }
 
+        private bool SupportsSearch
+        {
+            get
+            {
+                var capabilities = _capabilitiesProvider.GetCapabilities(Settings);
+
+                return capabilities.SupportedSearchParameters != null
+                    && capabilities.SupportedSearchParameters.Contains("q");
+            }
+        }
+
+        private bool SupportsTvSearch
+        {
+            get
+            {
+                var capabilities = _capabilitiesProvider.GetCapabilities(Settings);
+
+                return capabilities.SupportedTvSearchParameters != null
+                    && capabilities.SupportedTvSearchParameters.Contains("q")
+                    && capabilities.SupportedTvSearchParameters.Contains("season")
+                    && capabilities.SupportedTvSearchParameters.Contains("ep");
+            }
+        }
+
+        private bool SupportsTvdbSearch
+        {
+            get
+            {
+                var capabilities = _capabilitiesProvider.GetCapabilities(Settings);
+
+                return capabilities.SupportedTvSearchParameters != null
+                    && capabilities.SupportedTvSearchParameters.Contains("tvdbid")
```

Comment:
It's what we typically do, but in this case leading && seems clearer.
(Btw this was copied from TorznabRequestGenerator, not new code).